nanoFramework.Tools.DebugLibrary.Shared/NFDevice/GlobalExclusiveDeviceAccess.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using nanoFramework.Tools.Debugger.PortTcpIp;
+
+namespace nanoFramework.Tools.Debugger.NFDevice
+{
+    /// <summary>
+    /// Code that wants to access a device should use this system-wide exclusive access while
+    /// communicating to a device to prevent that another nanoFramework tool also wants to
+    /// communicate with the device.
+    /// </summary>
+    public static class GlobalExclusiveDeviceAccess
+    {
+        #region Fields
+        /// <summary>
+        ///  Base name for the system-wide mutex that controls access to a device connected to a COM port.
+        /// </summary>
+        private const string MutexBaseName = "276545121198496AADD346A60F14EF8D_";
```

Comment:
I see you finishes the string with underscore and while creating it, you also add another underscore. Is that wanted?

Reply:
No, once is enough. Fixed that.